e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/add-to-an-existing-payment.spec.js

```diff
@@ -149,4 +153,84 @@ context(`${PDC_TEAMS.PDC_RECONCILE} users can add fee records to existing paymen
       pages.utilisationReportAddToAnExistingPaymentPage.errorSummary().contains('Select a payment to add the fee or fees to');
     });
   });
+
+  it('should add the selected fees to the payments and update the fee record status after clicking the continue button', () => {
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+    cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord]);
+
+    navigateToAddToExistingPaymentScreenForFirstFeeRecord();
+
+    pages.utilisationReportAddToAnExistingPaymentPage.continueButton().click();
+
+    pages.utilisationReportPage.premiumPaymentsTab.getPaymentLink(PAYMENT_ID_ONE).should('contain', 'GBP 450.00');
+    pages.utilisationReportPage.premiumPaymentsTab.getPremiumPaymentsTableRow(FEE_RECORD_ID_ONE).should('contain', 'MATCH');
+  });
+
+  it('should redirect the user to then premium payments page when user clicks the back or cancel links and persist the selected fees', () => {
+    const FEE_RECORD_ID_THREE = '33';
+    const FEE_RECORD_ID_FOUR = '44';
+    const thirdFeeRecord = FeeRecordEntityMockBuilder.forReport(report)
+      .withId(FEE_RECORD_ID_THREE)
+      .withFacilityId('33333333')
+      .withExporter('Exporter 3')
+      .withPaymentCurrency(PAYMENT_CURRENCY)
+      .withFeesPaidToUkefForThePeriod(300)
+      .withFeesPaidToUkefForThePeriodCurrency('EUR')
+      .withPaymentExchangeRate(0.5)
+      .withStatus('TO_DO')
+      .build();
+    const fourthFeeRecord = FeeRecordEntityMockBuilder.forReport(report)
+      .withId(FEE_RECORD_ID_FOUR)
+      .withFacilityId('44444444')
+      .withExporter('Exporter 4')
+      .withPaymentCurrency(PAYMENT_CURRENCY)
+      .withFeesPaidToUkefForThePeriod(400)
+      .withFeesPaidToUkefForThePeriodCurrency('EUR')
+      .withPaymentExchangeRate(0.5)
+      .withPayments([firstPayment])
+      .withStatus('DOES_NOT_MATCH')
+      .build();
+
+    const navigateToAddToExistingPaymentScreen = () => {
+      cy.get('[type="submit"]').contains('Add a payment').click();
+      cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}/add-payment`));
+
+      cy.get('[type="submit"]').contains('Add reported fees to an existing payment').click();
+      cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}/add-to-an-existing-payment`));
+    };
+
+    const verifyExpectedFeeRecordCheckboxesAreChecked = () => {
+      getFeeRecordCheckbox([FEE_RECORD_ID_ONE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).should('be.checked');
+      getFeeRecordCheckbox([FEE_RECORD_ID_THREE], PAYMENT_CURRENCY, FEE_RECORD_STATUS.TO_DO).should('be.checked');
+      getFeeRecordCheckbox([FEE_RECORD_ID_TWO, FEE_RECORD_ID_FOUR], PAYMENT_CURRENCY, FEE_RECORD_STATUS.DOES_NOT_MATCH).should('not.be.checked');
+    };
```

Comment:
I'm not sure I like these defined inside the test itself but that might just be me. I think personally I would put the navigate function at the top of the file (might require rebalancing the current navigate helper) at the top of the file, and then I would just repeat the assertions since they are the most crucial part of the test.

Reply:
Repeated assertions, RE the navigate function the difference between the `navigateToAddToExistingPaymentScreenForFirstFeeRecord` and `navigateToAddToExistingPaymentScreen` helpers (for the shared lines) is the singular vs plural 'fees' when we get the 'Add reported fee to an existing payment' button in the following line:

`cy.get('[type="submit"]').contains('Add reported **fee** to an existing payment').click();`

We could use the `data-cy` attribute to get around this (so we could have one generic helper function to cover all the test cases), but we wouldn't be testing the button test on the e2e journey then. Any thoughts?

Reply:
Kept the `contains` approach, also kept `navigateToAddToExistingPaymentScreenForFirstFeeRecord` as I feel like it does it quite well for the 'first fee record' selected case, moved the `navigateToAddToExistingPaymentScreen` helper bits inline